### ISSUE TITLE
Videos crashing when in Full Screen mode - Intel Mac, 12.2.1

### ISSUE THREAD

**Issue by Andre-X86** [iina/iina#3650]
**System and IINA version:**

- macOS 12.2.1
- IINA 1.2.0

**Expected behavior:**
I just started using IINA and when playing MKV files the player will sometimes crash at the exact spot of a video when its in full screen mode. 

**Actual behavior:**

<details>
<summary><b>Crash report:</b></summary>

```
rocess:               IINA [14318]
Path:                  /Applications/IINA.app/Contents/MacOS/IINA
Identifier:            com.colliderli.iina
Version:               1.2.0 (129)
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2022-02-25 13:51:14.3360 -0700
OS Version:            macOS 12.2.1 (21D62)
Report Version:        12
Bridge OS Version:     6.2 (19P744)
Anonymous UUID:        A23BF958-FA1F-A9D1-7006-0E3949F64A39

Sleep/Wake UUID:       38FA5B77-7D23-4659-AEBC-0BE45B693B78

Time Awake Since Boot: 300000 seconds
Time Since Wake:       9503 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

ID   Vend/Dev
bac4 67ef1002
Seconds Ago   ID   Type
   430000.0   bac4 Attach
   430000.0   bac4 Eject
   430000.0   bac4 EjectFinalize
   430000.0   bac4 EjectFinalized
   430000.0   bac4 Detach


Application Specific Information:
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSNextStepFrame _displayName]: unrecognized selector sent to instance 0x7fdf52386fe0'
abort() called
terminating with uncaught exception of type NSException


Application Specific Backtrace 0:
0   CoreFoundation                      0x00007ff81b944f0b __exceptionPreprocess + 242
1   libobjc.A.dylib                     0x00007ff81b6a5b9d objc_exception_throw + 48
2   CoreFoundation                      0x00007ff81b9c9adb -[NSObject(NSObject) __retain_OA] + 0
3   CoreFoundation                      0x00007ff81b8abdb9 ___forwarding___ + 1406
4   CoreFoundation                      0x00007ff81b8ab7a8 _CF_forwarding_prep_0 + 120
5   AppKit                              0x00007ff81e4228f0 -[NSWindow title] + 140
6   AppKit                              0x00007ff81eb7d9b1 -[NSWindow _calcAndSetFilenameTitle] + 79
7   IINA                                0x000000010a0eeeb1 $s4IINA20MainWindowControllerC11updateTitleyyF + 785
8   IINA                                0x000000010a129b27 $s4IINA22PlayerWindowControllerC13windowDidLoadyyFy10Foundation12NotificationVcfU1_TA + 39
9   IINA                                0x000000010a1e0d29 $s10Foundation12NotificationVIegn_So14NSNotificationCIeyBy_TR + 105
10  Foundation                          0x00007ff81c703f0b -[__NSObserver _doit:] + 287
11  CoreFoundation                      0x00007ff81b8bef23 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
12  CoreFoundation                      0x00007ff81b95c3f9 ___CFXRegistrationPost_block_invoke + 49
13  Foundation                          0x00007ff81c7104a2 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
14  Foundation                          0x00007ff81c71039a -[NSBlockOperation main] + 98
15  Foundation                          0x00007ff81c710331 __NSOPERATION_IS_INVOKING_MAIN__ + 17
16  Foundation                          0x00007ff81c70f59f -[NSOperation start] + 785
17  Foundation                          0x00007ff81c70f275 __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 17
18  Foundation                          0x00007ff81c70f14a __NSOQSchedule_f + 182
19  libdispatch.dylib                   0x00007ff81b64aad8 _dispatch_call_block_and_release + 12
20  libdispatch.dylib                   0x00007ff81b64bcc9 _dispatch_client_callout + 8
21  libdispatch.dylib                   0x00007ff81b6585d8 _dispatch_main_queue_callback_4CF + 951
22  CoreFoundation                      0x00007ff81b907689 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
23  CoreFoundation                      0x00007ff81b8c875e __CFRunLoopRun + 2752
24  CoreFoundation                      0x00007ff81b8c75dd CFRunLoopRunSpecific + 563
25  HIToolbox                           0x00007ff8245044f1 RunCurrentEventLoopInMode + 292
26  HIToolbox                           0x00007ff824504247 ReceiveNextEventCommon + 587
27  HIToolbox                           0x00007ff824503fe5 _BlockUntilNextEventMatchingListInModeWithFilter + 70
28  AppKit                              0x00007ff81e2f6d88 _DPSNextEvent + 886
29  AppKit                              0x00007ff81e2f53f4 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1411
30  AppKit                              0x00007ff81e2e7919 -[NSApplication run] + 586
31  AppKit                              0x00007ff81e2bb7b7 NSApplicationMain + 816
32  IINA                                0x000000010a067429 main + 9
33  dyld                                0x000000010bacf4fe start + 462


Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	    0x7ff81b7cc112 __pthread_kill + 10
1   libsystem_pthread.dylib       	    0x7ff81b802214 pthread_kill + 263
2   libsystem_c.dylib             	    0x7ff81b74ed10 abort + 123
3   libc++abi.dylib               	    0x7ff81b7bf0b2 abort_message + 241
4   libc++abi.dylib               	    0x7ff81b7b01fd demangling_terminate_handler() + 266
5   libobjc.A.dylib               	    0x7ff81b6ad509 _objc_terminate() + 96
6   libc++abi.dylib               	    0x7ff81b7be4d7 std::__terminate(void (*)()) + 8
7   libc++abi.dylib               	    0x7ff81b7be488 std::terminate() + 56
8   libdispatch.dylib             	    0x7ff81b64bcdd _dispatch_client_callout + 28
9   libdispatch.dylib             	    0x7ff81b6585d8 _dispatch_main_queue_callback_4CF + 951
10  CoreFoundation                	    0x7ff81b907689 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
11  CoreFoundation                	    0x7ff81b8c875e __CFRunLoopRun + 2752
12  CoreFoundation                	    0x7ff81b8c75dd CFRunLoopRunSpecific + 563
13  HIToolbox                     	    0x7ff8245044f1 RunCurrentEventLoopInMode + 292
14  HIToolbox                     	    0x7ff824504247 ReceiveNextEventCommon + 587
15  HIToolbox                     	    0x7ff824503fe5 _BlockUntilNextEventMatchingListInModeWithFilter + 70
16  AppKit                        	    0x7ff81e2f6d88 _DPSNextEvent + 886
17  AppKit                        	    0x7ff81e2f53f4 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1411
18  AppKit                        	    0x7ff81e2e7919 -[NSApplication run] + 586
19  AppKit                        	    0x7ff81e2bb7b7 NSApplicationMain + 816
20  IINA                          	       0x10a067429 main + 9
21  dyld                          	       0x10bacf4fe start + 462

Thread 1:: mpv/mpv core
0   libsystem_kernel.dylib        	    0x7ff81b7c8506 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff81b802a69 _pthread_cond_wait + 1224
2   libmpv.1.dylib                	       0x10d37fd09 mp_dispatch_queue_process + 345
3   libmpv.1.dylib                	       0x10d3bc6ce run_playloop + 2958
4   libmpv.1.dylib                	       0x10d3b286c mp_play_files + 6860
5   libmpv.1.dylib                	       0x10d39822e core_thread + 62
6   libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
7   libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 2:: mpv/lua script (ytdl_hook)
0   libsystem_kernel.dylib        	    0x7ff81b7c8506 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff81b802a69 _pthread_cond_wait + 1224
2   libmpv.1.dylib                	       0x10d397a0f wait_wakeup + 95
3   libmpv.1.dylib                	       0x10d3983e5 mpv_wait_event + 357
4   libmpv.1.dylib                	       0x10d3b43f6 script_wait_event + 102
5   liblua.5.1.dylib              	       0x10bff3ab6 luaD_precall + 455
6   liblua.5.1.dylib              	       0x10bffd10c luaV_execute + 3514
7   liblua.5.1.dylib              	       0x10bff3f19 luaD_call + 97
8   liblua.5.1.dylib              	       0x10bff0550 lua_call + 41
9   libmpv.1.dylib                	       0x10d3b3f53 load_scripts + 291
10  liblua.5.1.dylib              	       0x10bff3ab6 luaD_precall + 455
11  liblua.5.1.dylib              	       0x10bff3f08 luaD_call + 80
12  liblua.5.1.dylib              	       0x10bff364f luaD_rawrunprotected + 84
13  liblua.5.1.dylib              	       0x10bff4180 luaD_pcall + 68
14  liblua.5.1.dylib              	       0x10bff05db lua_pcall + 105
15  libmpv.1.dylib                	       0x10d3b3aeb run_lua + 1307
16  liblua.5.1.dylib              	       0x10bff3ab6 luaD_precall + 455
17  liblua.5.1.dylib              	       0x10bff3f08 luaD_call + 80
18  liblua.5.1.dylib              	       0x10bff364f luaD_rawrunprotected + 84
19  liblua.5.1.dylib              	       0x10bff4180 luaD_pcall + 68
20  liblua.5.1.dylib              	       0x10bff063f lua_cpcall + 45
21  libmpv.1.dylib                	       0x10d3b3533 load_lua + 147
22  libmpv.1.dylib                	       0x10d3beba2 script_thread + 114
23  libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
24  libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 3:: mpv/lua script (stats)
0   libsystem_kernel.dylib        	    0x7ff81b7c8506 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff81b802a69 _pthread_cond_wait + 1224
2   libmpv.1.dylib                	       0x10d397a0f wait_wakeup + 95
3   libmpv.1.dylib                	       0x10d3983e5 mpv_wait_event + 357
4   libmpv.1.dylib                	       0x10d3b43f6 script_wait_event + 102
5   liblua.5.1.dylib              	       0x10bff3ab6 luaD_precall + 455
6   liblua.5.1.dylib              	       0x10bffd10c luaV_execute + 3514
7   liblua.5.1.dylib              	       0x10bff3f19 luaD_call + 97
8   liblua.5.1.dylib              	       0x10bff0550 lua_call + 41
9   libmpv.1.dylib                	       0x10d3b3f53 load_scripts + 291
10  liblua.5.1.dylib              	       0x10bff3ab6 luaD_precall + 455
11  liblua.5.1.dylib              	       0x10bff3f08 luaD_call + 80
12  liblua.5.1.dylib              	       0x10bff364f luaD_rawrunprotected + 84
13  liblua.5.1.dylib              	       0x10bff4180 luaD_pcall + 68
14  liblua.5.1.dylib              	       0x10bff05db lua_pcall + 105
15  libmpv.1.dylib                	       0x10d3b3aeb run_lua + 1307
16  liblua.5.1.dylib              	       0x10bff3ab6 luaD_precall + 455
17  liblua.5.1.dylib              	       0x10bff3f08 luaD_call + 80
18  liblua.5.1.dylib              	       0x10bff364f luaD_rawrunprotected + 84
19  liblua.5.1.dylib              	       0x10bff4180 luaD_pcall + 68
20  liblua.5.1.dylib              	       0x10bff063f lua_cpcall + 45
21  libmpv.1.dylib                	       0x10d3b3533 load_lua + 147
22  libmpv.1.dylib                	       0x10d3beba2 script_thread + 114
23  libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
24  libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 4:: mpv/lua script (console)
0   libsystem_kernel.dylib        	    0x7ff81b7c8506 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff81b802a69 _pthread_cond_wait + 1224
2   libmpv.1.dylib                	       0x10d397a0f wait_wakeup + 95
3   libmpv.1.dylib                	       0x10d3983e5 mpv_wait_event + 357
4   libmpv.1.dylib                	       0x10d3b43f6 script_wait_event + 102
5   liblua.5.1.dylib              	       0x10bff3ab6 luaD_precall + 455
6   liblua.5.1.dylib              	       0x10bffd10c luaV_execute + 3514
7   liblua.5.1.dylib              	       0x10bff3f19 luaD_call + 97
8   liblua.5.1.dylib              	       0x10bff0550 lua_call + 41
9   libmpv.1.dylib                	       0x10d3b3f53 load_scripts + 291
10  liblua.5.1.dylib              	       0x10bff3ab6 luaD_precall + 455
11  liblua.5.1.dylib              	       0x10bff3f08 luaD_call + 80
12  liblua.5.1.dylib              	       0x10bff364f luaD_rawrunprotected + 84
13  liblua.5.1.dylib              	       0x10bff4180 luaD_pcall + 68
14  liblua.5.1.dylib              	       0x10bff05db lua_pcall + 105
15  libmpv.1.dylib                	       0x10d3b3aeb run_lua + 1307
16  liblua.5.1.dylib              	       0x10bff3ab6 luaD_precall + 455
17  liblua.5.1.dylib              	       0x10bff3f08 luaD_call + 80
18  liblua.5.1.dylib              	       0x10bff364f luaD_rawrunprotected + 84
19  liblua.5.1.dylib              	       0x10bff4180 luaD_pcall + 68
20  liblua.5.1.dylib              	       0x10bff063f lua_cpcall + 45
21  libmpv.1.dylib                	       0x10d3b3533 load_lua + 147
22  libmpv.1.dylib                	       0x10d3beba2 script_thread + 114
23  libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
24  libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 5:: AMCP Logging Spool
0   libsystem_kernel.dylib        	    0x7ff81b7c5af6 semaphore_wait_trap + 10
1   caulk                         	    0x7ff82428c1c0 caulk::mach::semaphore::wait_or_error() + 16
2   caulk                         	    0x7ff82427308a caulk::concurrent::details::worker_thread::run() + 36
3   caulk                         	    0x7ff824272d4e void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*> > >(void*) + 41
4   libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
5   libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 6:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	    0x7ff81b7c5aba mach_msg_trap + 10
1   libsystem_kernel.dylib        	    0x7ff81b7c5e2b mach_msg + 59
2   CoreFoundation                	    0x7ff81b8c9af2 __CFRunLoopServiceMachPort + 319
3   CoreFoundation                	    0x7ff81b8c81cb __CFRunLoopRun + 1325
4   CoreFoundation                	    0x7ff81b8c75dd CFRunLoopRunSpecific + 563
5   AppKit                        	    0x7ff81e464fd8 _NSEventThread + 132
6   libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
7   libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 7:: CVDisplayLink
0   libsystem_kernel.dylib        	    0x7ff81b7c8506 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff81b802a9f _pthread_cond_wait + 1278
2   CoreVideo                     	    0x7ff8230487a6 CVDisplayLink::waitUntil(unsigned long long) + 366
3   CoreVideo                     	    0x7ff82304791a CVDisplayLink::runIOThread() + 534
4   libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
5   libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 8:: mpv/demux
0   libsystem_kernel.dylib        	    0x7ff81b7c8506 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff81b802a69 _pthread_cond_wait + 1224
2   libmpv.1.dylib                	       0x10d349fbf demux_thread + 223
3   libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
4   libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 9:: mpv/worker
0   libsystem_kernel.dylib        	    0x7ff81b7c8506 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff81b802a69 _pthread_cond_wait + 1224
2   libmpv.1.dylib                	       0x10d381a4c worker_thread + 76
3   libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
4   libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 10:: mpv/vo
0   libsystem_kernel.dylib        	    0x7ff81b7c8506 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff81b802a69 _pthread_cond_wait + 1224
2   libmpv.1.dylib                	       0x10d409886 vo_thread + 2054
3   libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
4   libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 11:: com.apple.coremedia.rootQueue.47
0   libsystem_kernel.dylib        	    0x7ff81b7c5b0e semaphore_timedwait_trap + 10
1   libdispatch.dylib             	    0x7ff81b64c1f2 _dispatch_sema4_timedwait + 72
2   libdispatch.dylib             	    0x7ff81b64c61f _dispatch_semaphore_wait_slow + 58
3   libdispatch.dylib             	    0x7ff81b65b1e7 _dispatch_worker_thread + 308
4   libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
5   libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 12:: com.apple.audio.IOThread.client
0   libsystem_kernel.dylib        	    0x7ff81b7c5aba mach_msg_trap + 10
1   libsystem_kernel.dylib        	    0x7ff81b7c5e2b mach_msg + 59
2   CoreAudio                     	    0x7ff81d44995b HALB_MachPort::SendSimpleMessageWithSimpleReply(unsigned int, unsigned int, int, int&, bool, unsigned int) + 111
3   CoreAudio                     	    0x7ff81d2d80ff HALC_ProxyIOContext::IOWorkLoop() + 3937
4   CoreAudio                     	    0x7ff81d2d6bc1 invocation function for block in HALC_ProxyIOContext::HALC_ProxyIOContext(unsigned int, unsigned int) + 63
5   CoreAudio                     	    0x7ff81d49f76c HALB_IOThread::Entry(void*) + 72
6   libsystem_pthread.dylib       	    0x7ff81b8024f4 _pthread_start + 125
7   libsystem_pthread.dylib       	    0x7ff81b7fe00f thread_start + 15

Thread 13::  Dispatch queue: com.colliderli.iina.controller
0   libsystem_kernel.dylib        	    0x7ff81b7c8506 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff81b802a69 _pthread_cond_wait + 1224
2   Foundation                    	    0x7ff81c71fa62 -[NSOperation waitUntilFinished] + 697
3   CoreFoundation                	    0x7ff81b8908a8 _CFXNotificationPost + 847
4   IINA                          	       0x10a1b1dde PlayerCore.postNotification(_:) + 190
5   IINA                          	       0x10a1d50cb MPVController.handlePropertyChange(_:_:) + 4523
6   IINA                          	       0x10a1d2c8c MPVController.handleEvent(_:) + 1564
7   IINA                          	       0x10a1d9d0a partial apply for closure #1 in MPVController.readEvents() + 42
8   IINA                          	       0x10a1cd9de thunk for @escaping @callee_guaranteed () -> () + 14
9   libdispatch.dylib             	    0x7ff81b64aad8 _dispatch_call_block_and_release + 12
10  libdispatch.dylib             	    0x7ff81b64bcc9 _dispatch_client_callout + 8
11  libdispatch.dylib             	    0x7ff81b651cee _dispatch_lane_serial_drain + 696
12  libdispatch.dylib             	    0x7ff81b6527c8 _dispatch_lane_invoke + 366
13  libdispatch.dylib             	    0x7ff81b65c7e1 _dispatch_workloop_worker_thread + 758
14  libsystem_pthread.dylib       	    0x7ff81b7ff074 _pthread_wqthread + 326
15  libsystem_pthread.dylib       	    0x7ff81b7fdffb start_wqthread + 15

Thread 14::  Dispatch queue: com.colliderli.iina.mpvgl
0   libsystem_kernel.dylib        	    0x7ff81b7c8506 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff81b802a69 _pthread_cond_wait + 1224
2   libmpv.1.dylib                	       0x10d40bd5b mpv_render_context_render + 907
3   IINA                          	       0x10a17076a specialized closure #1 in closure #1 in ViewLayer.draw(inCGLContext:pixelFormat:forLayerTime:displayTime:) + 154
4   IINA                          	       0x10a16f185 closure #1 in ViewLayer.draw(inCGLContext:pixelFormat:forLayerTime:displayTime:) + 133
5   IINA                          	       0x10a1709bd specialized ViewLayer.draw(inCGLContext:pixelFormat:forLayerTime:displayTime:) + 557
6   IINA                          	       0x10a16f1e1 @objc ViewLayer.draw(inCGLContext:pixelFormat:forLayerTime:displayTime:) + 33
7   QuartzCore                    	    0x7ff822a051bd CAOpenGLLayerDraw(CAOpenGLLayer*, double, CVTimeStamp const*, unsigned int) + 2021
8   QuartzCore                    	    0x7ff822a047f8 -[CAOpenGLLayer _display] + 585
9   IINA                          	       0x10a16f4fa @objc ViewLayer.display() + 58
10  IINA                          	       0x10a16f261 ViewLayer.draw(forced:) + 97
11  IINA                          	       0x10a1db3c2 partial apply for closure #1 in mpvUpdateCallback(_:) + 18
12  IINA                          	       0x10a1cd9de thunk for @escaping @callee_guaranteed () -> () + 14
13  libdispatch.dylib             	    0x7ff81b64aad8 _dispatch_call_block_and_release + 12
14  libdispatch.dylib             	    0x7ff81b64bcc9 _dispatch_client_callout + 8
15  libdispatch.dylib             	    0x7ff81b651cee _dispatch_lane_serial_drain + 696
16  libdispatch.dylib             	    0x7ff81b6527c8 _dispatch_lane_invoke + 366
17  libdispatch.dylib             	    0x7ff81b65c7e1 _dispatch_workloop_worker_thread + 758
18  libsystem_pthread.dylib       	    0x7ff81b7ff074 _pthread_wqthread + 326
19  libsystem_pthread.dylib       	    0x7ff81b7fdffb start_wqthread + 15

Thread 15:
0   libsystem_pthread.dylib       	    0x7ff81b7fdfec start_wqthread + 0

```
</details>

<br>

<details>
<summary><b>mpv log:</b></summary>

```
```
</details>

<br>

**Steps to reproduce:**

- [ ] MPV does not have this problem.

**How often does this happen?**

**Comment by Andre-X86** [comment]
The main feature that I love about this app is the ability to full screen while still being able to work on other screens. I hope a fix or workaround is available.

**Comment by Andre-X86** [comment]
I just started using IINA but have been experiencing crashes since using it. The videos play fine as a long as they are not full screen'd and the video will crash at the exact same spot when it is full screen'd.

**Comment by low-batt** [comment]
In IINA preferences under General is "Use legacy full screen" enabled?

I ask because this failure:
```text
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSNextStepFrame _displayName]: unrecognized selector sent to instance 0x7fa50f4d4530'
```

Matches the crash seen in issue #3543. Disabling legacy full screen works as a workaround. If you don't have legacy full screen enabled then I'm surprised as we have only seen this problem when that feature is enabled and macOS 11 or 12 is being used.

As it turns out Apple just responded to feedback filed about this failure requesting we test 12.3 Beta 2. Testing showed the problem no longer reproduced. It looks like this one will be fixed when macOS 12.3 is released.

For macOS 11-12.2 a workaround in the code has been merged. A release is in progress, but additional code reviews and merging need to be completed before the next release can be cut.

**Comment by Andre-X86** [comment]
Thanks for replying and the info provided. Disabling legacy full screen works but then I can't use the video player like i'm intending to. I'll wait for 12.3 and see what the results are at that point.

**Comment by low-batt** [comment]
A fix in PR #3547 has been merged into the develop branch.

See issue #3543 for details.

**Comment by low-batt** [comment]
Closing. Fixed in IINA 1.3.0.
